### PR TITLE
Feature/socket

### PR DESCRIPTION
**Pull request by condef5**
### This pull request includes:

* Add socket integration  with hook
* Add function `sendData`, this function allow send data through web socket
* Render list messages.
* Add new messages

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -21,14 +21,55 @@ const Container = styled.div`
     "channels new-message";
 `;
 
-function App() {
+function App({ url = "ws://localhost:4000" }) {
+  const ws = React.useRef(null);
```
Comment (CesarCachay) [inline client/src/App.js]: Could you explain please how does `useRef` work?

File: client/src/WriteMessage.js [hunk 2]
```diff
@@ -20,15 +20,23 @@ const hiddenSubmit = css`
   display: none;
 `;
 
-function WriteMessage() {
+function WriteMessage({ sendData }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const content = event.target.elements.message.value;
```
Comment (CesarCachay) [inline client/src/WriteMessage.js]: Good job! Why do you think it's necessary to use "elements" in this line?

File: client/src/App.js [hunk 3]
```diff
@@ -21,14 +21,55 @@ const Container = styled.div`
...
+        console.log("open");
+        setConnected(true);
+      };
+      ws.current.onclose = () => {
+        console.log("close");
+        setConnected(false);
+      };
+      ws.current.onmessage = ({ data }) => {
+        setMessages((state) => {
+          return [...state, JSON.parse(data)];
```
Comment (MariethPerez) [inline client/src/App.js]: Good job!!! Could you explain me please how '...' works?

File: client/src/App.js (hunk 1 above)
Comment (condef5) [reply client/src/App.js]: I use [refs](https://reactjs.org/docs/refs-and-the-dom.html) to be able to work with the hooks :)

File: client/src/WriteMessage.js (hunk 2 above)
Comment (condef5) [reply client/src/WriteMessage.js]: Because the form only have a input :)

File: client/src/App.js (hunk 3 above)
Comment (condef5) [reply client/src/App.js]: Of course, this is the [spread operator](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Operadores/Spread_operator), and it expands the data and joins them with the new element that comes through websockets `data`.

Another way to do this is:
```js
return state.concat(JSON.parse(data))
```

**Pull request merged**